### DISCUSSION TITLE
Node Collision Prevention

### DISCUSSION THREAD

**Discussion by plc-dev** [bcakmakoglu/vue-flow#1627] [Ideas]
Hi. Currently, there seems to be no inbuilt option of preventing node collisions, so that they are not allowed to overlap. Is such a feature planned? I'd be willing to submit a Pull Request, if pointed in the right direction.
I was thinking of behavior along the lines of [this](https://jbaysolutions.github.io/vue-grid-layout/guide/07-prevent-collision.html), including a visual clue of what the last successful drag-position might be.

**Comment by bcakmakoglu** [comment]
No, this feature is not planned and it's not something that I would want to add to the library anyway.
You can build this sort of thing yourself *on top* of VueFlow.
VueFlow is not meant to solve specific use-cases like these, it's meant to give you the tools to do this yourself :)

I'd be open to add an *example* to the docs for such an implementation but I would not want to add extra maintenance work to the core for something like this.

**Comment by plc-dev** [comment]
@bcakmakoglu Thank you for the response, and sorry for my late reply.

I've since implemented the collision prevention for the nodes for the regular Drag-Events regarding the position of the nodes. As you mentioned, it is pretty straight forward to implement the behavior on top of VueFlow, as the [NodeDragEvent](https://vueflow.dev/typedocs/interfaces/NodeDragEvent.html) exposes a reference to the node.

To implement the same collision prevention for the `ResizeHandler` seems to be more cumbersome, as the [OnResizeStart](https://vueflow.dev/typedocs/interfaces/NodeResizerEmits.html)-event does not expose a reference to the node (or an ID). Currently, I identify the resized node through the `sourceElement` on the event-property of the `DragEvent`, select the child-container with the class `vue-flow__node` that matches the position and set the `width`- and `height`-properties manually.

This seems unnecessarily complicated. Am I missing something? If not, would you be willing to extend the Resize-Events, so that the interface matches the interface of [NodeDragEvent](https://vueflow.dev/typedocs/interfaces/NodeDragEvent.html)?

If you want I can put up a PR, that introduces that change, while keeping backwards compatibility if necessary (just keep the `ResizeParams`?).
I can also put up a PR for the mentioned example in the docs for the collision prevention if you are interested.

Thanks for your response in advance!
